core/shared/src/main/scala/scala/macros/internal/prettyprinters/Structure.scala

```diff
@@ -0,0 +1,122 @@
+package scala.macros.internal
+package prettyprinters
+
+@scala.annotation.implicitNotFound("don't know how to prettyprint structure of ${T}")
+trait Structure[T] {
```

Comment:
Interesting design, looks better then the current system. I need to experiment with it more.

It looks like the main advantage is you can mix rendering style.